algebra-core/src/fields/mod.rs

```diff
@@ -158,10 +160,30 @@ pub trait Field:
     }
 }
 
-/// A trait that defines parameters for a prime field.
-pub trait FpParameters: 'static + Send + Sync + Sized {
+/// A trait that defines parameters for a field that can be used for FFT.
+pub trait FftParameters: 'static + Send + Sync + Sized {
     type BigInt: BigInteger;
 
+    /// 2^s * t = MODULUS - 1 with t odd. This is the two-adicity of
+    /// `Self::MODULUS`.
+    const TWO_ADICITY: u32;
+
+    /// 2^s root of unity computed by GENERATOR^t
+    const ROOT_OF_UNITY: Self::BigInt;
```

Comment:
Can this just be `Self`?
```suggestion
    /// 2^s root of unity computed by GENERATOR^t
    const ROOT_OF_UNITY: Self;
```